### PR TITLE
[WIP] Add gpu specific requirements to pulsar qld gpu playbook

### PR DESCRIPTION
**Pull request by jlqfab**
Initial untested commits for configuring Nvidia GPU with docker support for Nectar pulsar-qld-gpu's.

### CODE REVIEW

File: one-offs/initial_nvidia_config_and_disable_cgroups_v2_playbook.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
+- hosts: all
```
Comment (cat-bro) [inline one-offs/initial_nvidia_config_and_disable_cgroups_v2_playbook.yml:1]: hosts: all?

File: roles/nvidia-ctk/tasks/main.yml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+- name: Install nvidia-container-toolkit
+  apt:
+    name: nvidia-container-toolkit
+    state: present
+
+- name: Configure the container runtime and restart Docker
+  shell:
+    nvidia-ctk runtime configure --runtime=docker
+  notify: restart docker
```
Comment (cat-bro) [inline roles/nvidia-ctk/tasks/main.yml:46]: is there a handler somewhere that will handle this?

File: one-offs/initial_nvidia_config_and_disable_cgroups_v2_playbook.yml (hunk 1 above)
Comment (jlqfab) [reply one-offs/initial_nvidia_config_and_disable_cgroups_v2_playbook.yml:1]: intention is to run with a `--limit` for testing, but should be defined properly, as you say.

File: roles/nvidia-ctk/tasks/main.yml (hunk 2 above)
Comment (jlqfab) [reply roles/nvidia-ctk/tasks/main.yml:46]: There will be. Thanks - forgot to add it.

**Comment by jlqfab** [comment]
@cat-bro the `nvidia-ctk` role in `pulsar-qld-gpu_playbook.yml` requires both Nvidia and Docker to already be installed. Nvidia configuration for MIG mode and reboots for kernel modules only need to be done once, but this needs to be done prior to the main playbook being run. Hence, addition of a check for nvidia at the start of the main playbook.

**Review by cat-bro — APPROVED** [review]

**Pull request merged**